File: client/src/app/shared/models/shoppingCart.ts
﻿import {nanoid} from 'nanoid';

export type Cart = {
  id: string;
  items: CartItem[];
  paymentIntentId?: string;
  clientSecret?: string;
  expirationTime?: Date;
}

export type CartItem = {
  campsiteId: number;
  campgroundId: number;
  campsiteName: string;
  campsiteType: string;
  campgroundName: string;
  startDate: Date;
  endDate: Date;
  price: number;
}

export class ShoppingCart implements Cart {
  id = 'cart_' + nanoid();
  items: CartItem[] = [];
  paymentIntentId?: string;
  clientSecret?: string;
  expirationTime?: Date;
}

File: Core/Models/ShoppingCart.cs
﻿namespace Core.Models;

public class ShoppingCart
{
    public required string Id { get; set; }
    public List<CartItem> Items { get; set; } = [];
    public string? ClientSecret { get; set; }
    public string? PaymentIntentId { get; set; }
    public DateTime ExpirationTime { get; set; }
}